implementations/rust/ockam/ockam_api/src/kafka/kafka_portal_worker.rs

```diff
@@ -0,0 +1,423 @@
+use crate::kafka::decoder::KafkaDecoder;
+use crate::kafka::encoder::KafkaEncoder;
+use crate::kafka::kafka_inlet_map::KafkaInletMap;
+use crate::kafka::protocol_aware::ProtocolState;
+use ockam::Context;
+use ockam_core::{
+    errcode::{Kind, Origin},
+    Address, AllowAll, Decodable, Encodable, Error, LocalInfo, LocalMessage, Route, Routed,
+    TransportMessage, Worker,
+};
+use ockam_transport_tcp::{PortalMessage, MAX_PAYLOAD_SIZE};
+use std::sync::Arc;
+
+///by default kafka supports up to 1MB messages
+pub(crate) const MAX_KAFKA_MESSAGE_SIZE: usize = 16 * 1024 * 1024;
```

Comment:
Should it be 1MB or 16MB?

Reply:
16mb it's the maximum "sane" message size suggested, it'll add a note